src/http/client.rs

```diff
@@ -57,6 +58,11 @@ impl<'app> Client<'app> {
 
         let host = url.host_str().unwrap_or("invalid host");
 
+        // You have logic in your initialization to ensure that the allowed_hosts are valid.
+        // I would constrain this type further so that you couldn't in code write something like
+        // let a = vec!["invalid host".to_string()];
+        // let c = Client::new(&a);
+        // let b = c.allowed_hosts.contains(&a.get(1).unwrap().clone());
```

Comment:
What should I do to constrain this type further? Type `allowed_hosts` as an array of URI like you suggested below?

Reply:
That it is a fine approach.

Reply:
I ended up using `url::Host` for better type checking — let me know what you think: https://github.com/walterbm/rusty-resizer/commit/be7af520978499f98b8b9b0ccaf4225967f9d45f